src/datreant/core/collections.py

```diff
@@ -592,19 +594,23 @@ def __getitem__(self, index):
         # we can take lists of indices, names, or uuids; these return a
         # Bundle; repeats already not respected since Bundle functions as a
         # set
-        if ((isinstance(index, list) or hasattr(index, 'dtype')) and
-                all([isinstance(item, bool) for item in index])):
-            # boolean indexing
-            memberlist = self._list()
+        memberlist = self._list()
+
+        if ((isinstance(index, list) and all(isinstance(item, bool) for item in index))
+            or
+            (hasattr(index, 'dtype') and index.dtype == 'bool')):
```

Comment:
`index` is the thing we're slicing with, and the `all` will exit early as soon as it hits a non bool item, so I'm not sure order will matter?